packages/graphql/src/types/app.ts

```diff
@@ -66,65 +87,67 @@ export type QuestionOptionsNumerical = {
   }[]
 }
 
-export type QuestionOptionsFreeText = {
+export interface ElementOptionsFreeText extends BaseElementOptions {
   solutions?: string[]
   restrictions?: {
     maxLength?: number | null
   }
 }
 
-export type QuestionOptions =
-  | QuestionOptionsChoices
-  | QuestionOptionsNumerical
-  | QuestionOptionsFreeText
+export type ElementOptions =
+  | ElementOptionsChoices
+  | ElementOptionsNumerical
+  | ElementOptionsFreeText
+
+export interface BaseElementData {
+  type: ElementType
 
-export interface BaseQuestionData {
   id: number
   name: string
-  type: QuestionType
-  displayMode: QuestionDisplayMode
   content: string
   pointsMultiplier: number
-  explanation: string | null
+  explanation?: string | null
+
+  options: object
+
+  // TODO: these legacy props have been moved to options
+  displayMode: ElementDisplayMode
   hasSampleSolution: boolean
   hasAnswerFeedbacks: boolean
-  options: object
 }
 
-export type BaseQuestionDataKeys = (keyof BaseQuestionData)[]
+export type BaseElementDataKeys = (keyof BaseElementData)[]
 
-interface IQuestionData<
-  Type extends QuestionType,
-  Options extends QuestionOptions
-> extends Question {
+interface IElementData<Type extends ElementType, Options extends ElementOptions>
+  extends Element {
   type: Type
   options: Options
 }
 
-export type ChoicesQuestionData = IQuestionData<
+export type ChoicesElementData = IElementData<
   'SC' | 'MC' | 'KPRIM',
```

Comment:
Here we might get the first inconsistencies, if we also allow QUESTION_SC, etc. now as DB types. Probably it would make sense to type all of this consistently and migrate from SC to QUESTION_SC from the beginning?

Reply:
Also, do we have to use string values here or could we also use ElementType.Question_FREE_TEXT or something?

Reply:
nope. that will cause a typescript error, as using a type as a value is not possible. so we will have to work hardcoded here

Reply:
Ah that makes sense...